pkg/cmd/api/api.go

```diff
@@ -299,22 +299,38 @@ func apiRun(opts *ApiOptions) error {
 
 	template := export.NewTemplate(opts.IO, opts.Template)
 
+	var pages interface{}
+	var serverError string
+	buffer := &bytes.Buffer{}
+
+	isJSON := false
 	hasNextPage := true
 	for hasNextPage {
 		resp, err := httpRequest(httpClient, host, method, requestPath, requestBody, requestHeaders)
 		if err != nil {
 			return err
 		}
 
-		endCursor, err := processResponse(resp, opts, headersOutputStream, &template)
+		var endCursor string
+		isJSON, endCursor, serverError, err = processResponse(resp, opts, headersOutputStream, buffer)
 		if err != nil {
+			// Print the server error message before exiting
+			_ = printResponse(buffer, opts, isJSON, serverError != "", opts.IO.Out, &template)
```

Comment:
Last question regarding printing, with responses now getting merged together when paginating, what are your thoughts on how we display headers when then `--include` flag is also specified? I am wondering if it makes sense to just display the headers of the last response.

Reply:
I had looked into refactoring this all up but it ended up looking more monolithic or, alternatively, passing a bunch of context parameters through to a func. I'm certainly not opposed to it and can take another look, but open to ideas that are actually an overall improvement. Ultimately I wanted to just print responses as they come, but I suppose if we're not paging there's only one page to print anyway. Since it's been a while, perhaps I can go back into this with a fresh pair of eyes.

As for `--include`: that's a good point. I didn't even consider that case and will add a test for it (I'm a fan of TDD). What about printing that to stderr? Though, that would be a breaking change. Probably too late, but I do see other tools like `curl` do that for similar logging/debug messages.

I can see pros and cons to either the first or last response. Though, if the first response fails it is the last response so maybe that just works. Alternatively, I could collect them all up into a buffer and print them all relative to the printed response. But maybe that's TMI and the last response is suitable. Thoughts?

Reply:
While I like the idea of printing the headers to stderr I think you are right that would a breaking change that we shouldn't make at this time.

I also don't think it is necessary to print all the headers relative to the printed response. As for printing either the first or last, I think the last is slightly better as it will have more up to date rate limit headers incase the user is using the API heavily and looking at the headers to inform them of any rate limits they me bumping up against.

Reply:
Will do. I also think the fact that last == first in case of an error also makes the last response headers ideal for both reasons. I got a good start on this last night but still working on refactoring to remove any unnecessary copies, still refactor some bits out so that method isn't so monolithic (maybe even breaking up `apiRun` itself a bit), etc.